app.py

```diff
@@ -147,13 +162,63 @@ def graph_refresh() -> None:
             return
 
 
+def graph_refresh_package_analyzer() -> None:
+    """Schedule refresh for packages that are not yet analyzed by package analyzer."""
+    packages = _GRAPH_DB.retrieve_unanalyzed_python_package_versions()
+
+    if not packages:
+        _LOGGER.info("No unanalyzed packages found")
+        return
+
+    count = 0
+    for item in packages:
+        package = item["package_name"]
+        version = item["package_version"]
+        url = item["index_url"]
+        try:
+            analysis_id = openshift.schedule_package_analyzer(
+                package_name=package,
+                package_version=version,
+                index_url=url,
+                output=_PACKAGE_ANALYZER_OUTPUT,
+            )
+        except Exception as ecx:
+            # If we get some errors from OpenShift master - do not retry. Rather schedule the remaining
+            # ones and try to schedule the given package in the next run.
+            _LOGGER.exception(
+                f"Failed to schedule new package analyzer to analyzer package {package} in version {version}"
+                f"from {url}, the graph refresh job will not fail but will try to reschedule this in next run"
+            )
+            _METRIC_PACKAGE_ANALYZERS_UNSCHEDULED.labels(package).inc()
+            continue
+
+        _LOGGER.info(
+            "Scheduled package analyzer for package %r, version %r, index_url %r, analysis is %r",
+            ,
+            package,
+            version,
+            url,
+            analysis_id,
+        )
+        _METRIC_PACKAGE_ANALYZERS_SCHEDULED.labels(package).inc()
+
+        count += 1
+        if _THOTH_GRAPH_REFRESH_EAGER_STOP and count >= _THOTH_GRAPH_REFRESH_EAGER_STOP:
+            _LOGGER.info(
+                "Eager stop of scheduling new package analyzer runs for unanalyzed packages, packages scheduled: %d",
+                count,
+            )
+            return
+
+
 def main():
     """Perform graph refresh job."""
     _LOGGER.info(f"Version v{__version__}")
     _LOGGER.debug("Debug mode is on")
 
     with _METRIC_RUNTIME.time():
-        graph_refresh()
+        graph_refresh_solver()
+        graph_refresh_package_analyzer()
 
     if _THOTH_METRICS_PUSHGATEWAY_URL:
         try:
```

Comment:
Starting from this I did no changes.

Reply:
@fridex I think this is how you wanted me to do, right?

Reply:
Yes! :1st_place_medal:

Reply:
Thanks